### PR TITLE
[Tests only][aptos_framework] chain_id: initialize for test (to specify any chain_id)

### PR DESCRIPTION
**Pull request by linnefromice**
### Description

enable to configure any chain_id in test.
to implement different processes depending on the environment judged by chain_id.
e.g. if make dummy value available in testnet, devnet (not mainnet)

### Test Plan
```bash
[ PASS    ] 0x1::chain_id::test_get # check that user can change id and get it
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4406)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/chain_id.move [hunk 1]
```diff
@@ -21,4 +21,18 @@ module aptos_framework::chain_id {
     public fun get(): u8 acquires ChainId {
         borrow_global<ChainId>(@aptos_framework).id
     }
+
+    #[test_only]
+    fun change_id(value: u8) acquires ChainId {
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/chain_id.move:26]: This should be public or you won't be able to access this anyway :)

File: aptos-move/framework/aptos-framework/sources/chain_id.move [hunk 2]
```diff
@@ -21,4 +21,18 @@ module aptos_framework::chain_id {
     public fun get(): u8 acquires ChainId {
         borrow_global<ChainId>(@aptos_framework).id
     }
+
+    #[test_only]
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/chain_id.move:25]: Why would you ever need to change the chain id in move unit tests? May be you're looking for initialize_for_test so you can initialize the chain id once?

**Review by movekevin — CHANGES_REQUESTED** [review]
Why do you need to change

File: aptos-move/framework/aptos-framework/sources/chain_id.move (hunk 1 above)
Comment (linnefromice) [reply aptos-move/framework/aptos-framework/sources/chain_id.move]: Sorry, I was careless.

File: aptos-move/framework/aptos-framework/sources/chain_id.move (hunk 2 above)
Comment (linnefromice) [reply aptos-move/framework/aptos-framework/sources/chain_id.move:25]: I don't have to be able to change it, but I would like to specify.
I agree with you, "initialize_for_test" is enough. I fixed. 2560a3e
At first I thought of this flow.
```txt
genesis::setup();
chain_id::test_get();
```

**Review by movekevin — APPROVED** [review]

**Comment by linnefromice** [comment]
@movekevin @wrwg additional review, please.

**Review by gregnazario — APPROVED** [review]

**Comment by linnefromice** [comment]
When will they be merged?

**Pull request merged**